### PR TITLE
Fix potential segfault with many BCs, refactor

### PR DESCRIPTION
**Pull request by flomnes**

### CODE REVIEW

File: src/solver/variable/surveyresults/surveyresults.cpp [hunk 1]
```diff
@@ -562,24 +563,40 @@ static inline void WriteIndexHeaderToFileDescriptor(int precisionLevel,
...
+      - Production
+      - NODU,
+      - NP Costs
+      - Net profit
+    */
+    const uint nbVariablesPerDetailRenewableCluster = 1;
+    // Production
+
+    const uint nbVariablesPerInequalityBindingConstraint = 1;
+    // Marginal price
```
Comment (flomnes) [inline src/solver/variable/surveyresults/surveyresults.cpp:586]: Determine at compilation ?

File: src/libs/antares/study/runtime/runtime.cpp [hunk 2]
```diff
@@ -661,6 +663,19 @@ void StudyRuntimeInfos::initializeMaxClusters(const Study& study)
       = maxNumberOfClusters<CompareAreasByNumberOfClusters::renewable>(study);
 }
 
+void StudyRuntimeInfos::initializeInequalityBindingConstraints()
```
Comment (guilpier-code) [inline src/libs/antares/study/runtime/runtime.cpp]: Do we initialize, here ? I guess we rather store inequality binding constraints numbers, do we not ?
If so, I suggest we call this method **getInequalityBindingConstraintsNumbers()**. 
We do not need to store these numbers, so we just have to return them (vector<uint>)

File: src/solver/variable/bindConstraints.hxx [hunk 3]
```diff
@@ -102,25 +100,10 @@ inline void BindingConstraints<NextT>::provideInformations(I& infos)
...
-        }
-    }
-    return bindConstGlobalNumbers;
-}
-
 template<class NextT>
 void BindingConstraints<NextT>::initializeFromStudy(Data::Study& study)
 {
-    std::vector<uint> InequalityBCnumbers = getInequalityBindConstraintGlobalNumbers(study);
+    const std::vector<uint>& InequalityBCnumbers = study.runtime->bindingConstraintGlobalNumbers;
```
Comment (guilpier-code) [inline src/solver/variable/bindConstraints.hxx]: I'm not at ease with getting a data directly in the runtime class like this.
According to remarks made in source file **runtime.cpp**, we can either have : 
- **getInequalityBindConstraintGlobalNumbers**(study) computing the global numbers and return a vector of them (which it not costly with the vector move constructor)
-  **getInequalityBindConstraintGlobalNumbers**(study) could return a reference to the the global numbers as a data member of runtime class

File: src/solver/variable/surveyresults/surveyresults.cpp [hunk 4]
```diff
@@ -562,24 +563,43 @@ static inline void WriteIndexHeaderToFileDescriptor(int precisionLevel,
...
+
+    const auto max = [](uint a, uint b, uint c, uint d) { return std::max({a, b, c, d}); };
+
+    return max(maxVars,
+               static_cast<uint>(nbVariablesPerDetailThermalCluster
+                                 * runtime->maxThermalClustersForSingleArea),
+               static_cast<uint>(nbVariablesPerDetailRenewableCluster
+                                 * runtime->maxRenewableClustersForSingleArea),
+               static_cast<uint>(nbVariablesPerInequalityBindingConstraint
+                                 * runtime->bindingConstraintGlobalNumbers.size()));
```
Comment (guilpier-code) [inline src/solver/variable/surveyresults/surveyresults.cpp]: Same remark as in **bindConstraints.hxx**

**Review by guilpier-code — CHANGES_REQUESTED** [review]

File: src/libs/antares/study/runtime/runtime.h [hunk 5]
```diff
@@ -175,6 +179,9 @@ class StudyRuntimeInfos
     //! The maximum number of renewable clusters for each area
     size_t maxRenewableClustersForSingleArea = 0;
 
+    //! The global indices for inequality ('<' or '>') binding constraints
+    std::vector<uint> bindingConstraintGlobalNumbers;
```
Comment (guilpier-code) [inline src/libs/antares/study/runtime/runtime.h]: **bindingConstraintGlobalNumbers** seems unused

**Review by guilpier-code — CHANGES_REQUESTED** [review]

**Comment by flomnes** [comment]
@guilpier-code DONE

**Comment by flomnes** [comment]
@guilpier-code DONE

**Review by guilpier-code — APPROVED** [review]

**Pull request merged**